ui/plugin-system/src/components/DatasourceEditorForm/DatasourceEditorForm.tsx

```diff
@@ -65,28 +66,42 @@ export function DatasourceEditorForm<T extends Datasource>(props: DatasourceEdit
     resolver: zodResolver(datasourceEditValidationSchema),
     mode: 'onBlur',
     defaultValues: {
-      name: state.metadata.name,
+      name: state.name,
       title: state.spec.display?.name,
       description: state.spec.display?.description,
       default: state.spec.default,
     },
   });
 
   const processForm: SubmitHandler<DatasourceEditValidationType> = () => {
-    onSave(state);
+    // reset display fields to undefined when empty (= undo the getInitialState patch)
+    const name = state.spec.display?.name;
+    const desc = state.spec.display?.description;
+    const finalDisplay: Display | undefined =
+      name != '' || desc != ''
```

Comment:
I think we could remove check about description and simplify
```suggestion
      const finalDisplay: Display | undefined = {...state.spec.display, name ? name : undefined}
```